### PR TITLE
Add _compute endpoint + _deprecate _range endpoint

### LINKED ISSUE #463: Add _compute endpoint
Through this endpoint it's possible to calculate, for a **given field**
- avg
- max
- min
- sum
- cardinality
- spanning

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
Fix #463 #464

### CODE REVIEW

File: arlas-core/src/main/java/io/arlas/server/model/request/Search.java [hunk 1]
```diff
@@ -19,8 +19,10 @@
 
 package io.arlas.server.model.request;
 
+import java.util.List;
```
Comment (alainbodiguel) [inline arlas-core/src/main/java/io/arlas/server/model/request/Search.java]: useless import?

File: docs/arlas-api-exploration.md [hunk 2]
```diff
@@ -11,10 +11,11 @@ The table below lists the URL endpoints and their optional "parts". A part is co
...
-| /arlas/explore/`{collection}`/**_geosearch**/`{z}`/`{x}`/`{y}`?`filter` & `form` & `projection` & `page`& `returned_geometries` | Search and return the elements found in the collection and localized in the given tile(x,y,z) as features, given the filters |
+| /arlas/explore/`{collection}`/**_range**?`compute` & `filter` & `form` | **DEPRECATED**. Calculates the min and max values of a field in the collection, given the filters |
+| /arlas/explore/`{collection}`/**_compute**?`compute` & `filter` & `form` | Computes the metric of a field in the collection, given the filters |
+| /arlas/explore/`{collection}`/**_search**?`filter` & `form` & `projection` & `page` | Search and return the elements found in the collection, given the filters |
+| /arlas/explore/`{collection}`/**_geosearch**?`filter` & `form` & `projection` & `page` | Search and return the elements found in the collection as features, given the filters |
+| /arlas/explore/`{collection}`/**_geosearch**/`{z}`/`{x}`/`{y}`?`filter` & `form` & `projection` & `page` | Search and return the elements found in the collection and localized in the given tile(x,y,z) as features, given the filters |
```
Comment (alainbodiguel) [inline docs/arlas-api-exploration.md]: returned_geometries is no longer there?

File: arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java [hunk 3]
```diff
@@ -147,6 +148,67 @@ public SearchResponse aggregate(MixedRequest request, CollectionReference collec
...
+        applyFilter(request.basicRequest.filter, fluidSearch);
+        applyFilter(request.headerRequest.filter, fluidSearch);
+        String field = ((ComputationRequest)request.basicRequest).field;
+        ComputationEnum metric = ((ComputationRequest)request.basicRequest).metric;
+        fluidSearch = fluidSearch.compute(field, metric);
+        SearchResponse response;
+        try {
+            response = fluidSearch.exec();
+        } catch (SearchPhaseExecutionException e) {
+            throw new InvalidParameterException("The field's type must be numeric");
```
Comment (sylvaingaudan) [inline arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java]: It seems strange to catch a generic exception and translate it into a very specific exception. We might miss some cases and report to the user an entirely unrelated error. Shouldn't we provide the e.getMessage() ?

File: arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java [hunk 4]
```diff
@@ -147,6 +148,67 @@ public SearchResponse aggregate(MixedRequest request, CollectionReference collec
...
+        }
+
+        ComputationResponse computationResponse = new ComputationResponse();
+        Long startQuery = System.nanoTime();
```
Comment (sylvaingaudan) [inline arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java]: startQueryTimestamp

File: arlas-core/src/main/java/io/arlas/server/model/request/Search.java (hunk 1 above)
Comment (MohamedHamouGisaia) [reply arlas-core/src/main/java/io/arlas/server/model/request/Search.java]: fixed

File: docs/arlas-api-exploration.md (hunk 2 above)
Comment (MohamedHamouGisaia) [reply docs/arlas-api-exploration.md]: fixed

File: arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java (hunk 3 above)
Comment (MohamedHamouGisaia) [reply arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java]: Fixed

**Pull request merged**